crawler_mcp/core/connection_pool.py

```
@@ -0,0 +1,320 @@
+                return
+
+            except TimeoutError:
+                if attempt < max_retries - 1:
+                    logger.warning(
+                        f"Timeout acquiring connection, attempt {attempt + 1}/{max_retries}"
+                    )
+                    await asyncio.sleep(retry_delay * (2**attempt))
+                else:
+                    raise RuntimeError(
+                        "Failed to acquire connection from pool after retries"
+                    )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)